### ISSUE TITLE
English version?

### ISSUE THREAD

**Issue by Tennoo** [Snailclimb/Java-Guide#18]
Hi! First of all, I just want to say that this is an amazing resource for learning Java. You have done a very good job.

I'm just wondering if an English version will ever be available? Or if translated versions of the pictures and diagrams would be available? I've currently been using Google translate and it works, but I'm just asking :)

Thank you for this!

**Comment by Snailclimb** [comment]
Your suggestion is very good.More than one foreign friend mentioned this suggestion.But this is a time-consuming and laborious task, and I may not be able to do it alone.If there are other people who will help with the translation later, you may have an English version. thank you for your support!

**Comment by donpaul120** [comment]
I think google automatically translate this to english. I actually do read this in english version through the web aside the images tho.

**Comment by Tennoo** [comment]
Yeah, I also use Google translate which translates the full page. Only having the images translated would be perfect.

**Comment by Adikso** [comment]
Why do you even add english only description to the repository. It is not the first time that I am interested and then see that everything is in fact in chinese despite english description :/

**Comment by charllie** [comment]
Learning Chinese is a core knowledge that most Java programmers need to master

**Comment by donpaul120** [comment]
@charllie Lol... I didn't know that until now, so all this years all i needed to do was to learn Chinese to harness the full power of JAVA... hmmmm

**Comment by feia1236** [comment]
So, I'm good at Chinese, but why not Java... hmmmm

**Comment by feia1236** [comment]
> @charllie So, I'm good at Chinese, but why not Java... hmmmm

**Comment by crazyqiang** [comment]
> Fully understandable. It's a lot to ask for, so I'm not expecting anything 😄
> 
> What about translating the pictures and such? Like this:
> ![image](https://user-images.githubusercontent.com/35228879/44954869-bab61000-aea9-11e8-8b41-e6c4079e8607.png)
> 
> Would be very helpful. Thank you again for this amazing contribution.

This tells us how important it is to learn Chinese well 😝

**Comment by jiaofanting** [comment]
哈哈哈哈哈，欢迎大家积极学习中文

**Comment by Randyzz** [comment]
让我想笑^_^

**Comment by itzdc** [comment]
洋人还用看练习题呢呀

**Comment by zmx104283** [comment]
看用户名感觉是个中国人

**Comment by unclehh** [comment]
hahaha

**Comment by huanghe2015** [comment]
Well,I think the pic is quite common in the theory of Opreation System's Textbook.
我认为这张图在操作系统原理的相关书籍中很容易找到。

**Comment by maxwellens** [comment]
> Fully understandable. It's a lot to ask for, so I'm not expecting anything 😄
> 
> What about translating the pictures and such? Like this:
> ![image](https://user-images.githubusercontent.com/35228879/44954869-bab61000-aea9-11e8-8b41-e6c4079e8607.png)
> 
> Would be very helpful. Thank you again for this amazing contribution.



> Fully understandable. It's a lot to ask for, so I'm not expecting anything 😄
> 
> What about translating the pictures and such? Like this:
> ![image](https://user-images.githubusercontent.com/35228879/44954869-bab61000-aea9-11e8-8b41-e6c4079e8607.png)
> 
> Would be very helpful. Thank you again for this amazing contribution.

用户：User
操作系统：Operating System
外壳：Shell
内核：Kernel
计算机硬件：Computer Hardware

**Comment by q258523454** [comment]
来我伟大的祖国找面经，点赞

**Comment by yangquanxyz** [comment]
> Fully understandable. It's a lot to ask for, so I'm not expecting anything 😄
> 
> What about translating the pictures and such? Like this:
> ![image](https://user-images.githubusercontent.com/35228879/44954869-bab61000-aea9-11e8-8b41-e6c4079e8607.png)
> 
> Would be very helpful. Thank you again for this amazing contribution.

maybe you can use the app named 'Baidu Translate',just choose which language you want to transfer to and taking a photo and it will translate the words in the pic to english.just like this
![WechatIMG19](https://user-images.githubusercontent.com/43701303/58321851-defbb780-7e51-11e9-8eed-5eef841613b4.jpeg)

**Comment by zebinh** [comment]
哈哈哈，好想笑啊，英文资料不是更多吗，不过这是面筋，外国友人没有写面筋的习惯吗？

**Comment by sanwancoder** [comment]
https://www.journaldev.com/java-tutorial-java-ee-tutorials# 

Perhaps, the website can help u

**Comment by jorksz** [comment]
six six six

**Comment by gdouzwt** [comment]
我可以翻译🤣

**Comment by adolhongping** [comment]
Learning English,not Java.

**Comment by butnotyou** [comment]
hhh  Surprise

**Comment by Lotus-Blue** [comment]
amazing gay

**Comment by tanjiuyang** [comment]
wo zhi xiang shuo zhen de hen bang,bu yao wen wo wei shen me ying yu zhe me hao

**Comment by timmmGZ** [comment]
six six six, how you learn Chinese already

**Comment by sukai-cheng** [comment]
作为中国人，我很骄傲
